### PR TITLE
Modularizing third party auth.

### PR DESCRIPTION
**Pull request by davidjoy**
NOTE: At present, I couldn't figure out how to test this in devstack.  That sucks.

### CODE REVIEW

File: src/account-settings/reducers.js [hunk 1]
```diff
@@ -223,6 +180,15 @@ const reducer = (state = defaultState, action) => {
...
 
+    case DISCONNECT_AUTH.BEGIN:
+    case DISCONNECT_AUTH.SUCCESS:
+    case DISCONNECT_AUTH.FAILURE:
+    case DISCONNECT_AUTH.RESET:
+      return {
+        ...state,
+        thirdPartyAuth: thirdPartyAuthReducer(state.thirdPartyAuth, action),
+      };
+
```
Comment (abutterworth) [inline src/account-settings/reducers.js]: Would `combineReducers` work here? https://redux.js.org/api/combinereducers

File: src/account-settings/third-party-auth/ThirdPartyAuth.jsx [hunk 2]
```diff
@@ -124,18 +123,22 @@ ThirdPartyAuth.propTypes = {
     connected: PropTypes.bool,
     id: PropTypes.string,
   })),
-  disconnectingState: PropTypes.objectOf(PropTypes.oneOf([null, 'pending', 'complete', 'error'])),
-  disconnectErrors: PropTypes.objectOf(PropTypes.bool),
+  providerStatuses: PropTypes.objectOf(PropTypes.oneOf([null, 'pending', 'complete', 'error'])),
```
Comment (abutterworth) [inline src/account-settings/third-party-auth/ThirdPartyAuth.jsx]: I think it's worth noting that these statuses are only for disconnection. Not sure if it's worth a comment or different name.

File: src/account-settings/reducers.js (hunk 1 above)
Comment (davidjoy) [reply src/account-settings/reducers.js]: Yes, that's the plan.

File: src/account-settings/third-party-auth/ThirdPartyAuth.jsx (hunk 2 above)
Comment (davidjoy) [reply src/account-settings/third-party-auth/ThirdPartyAuth.jsx]: Ahh, fair.  Hm.  Will rename. 👍

**Pull request merged**